src/Propel/Runtime/ActiveQuery/Criterion/ExistsCriterion.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/**
+ * MIT License. This file is part of the Propel package.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Propel\Runtime\ActiveQuery\Criterion;
+
+use Propel\Runtime\ActiveQuery\ModelJoin;
+use Propel\Runtime\Map\RelationMap;
+
+/**
+ * Specialized Criterion used for EXISTS
+ */
+class ExistsCriterion extends AbstractCriterion
+{
+    public const TYPE_EXISTS = 'EXISTS';
+
+    public const TYPE_NOT_EXISTS = 'NOT_EXISTS';
+
+    /**
+     * The inner query of the exists
+     *
+     * @var \Propel\Runtime\ActiveQuery\ModelCriteria
+     */
+    private $existsQuery;
+
+    /**
+     * Build NOT EXISTS instead of EXISTS
+     *
+     * @var string $keyword Either ExistsCriterion::TYPE_EXISTS or ExistsCriterion::TYPE_NOT_EXISTS
+     *
+     * @phpstan-var ExistsCriterion::TYPE_*
+     */
+    private $type = self::TYPE_EXISTS;
+
+    /**
+     * @phpstan-param ExistsCriterion::TYPE_*|null $type
+     *
+     * @param \Propel\Runtime\ActiveQuery\ModelCriteria $outerQuery
+     * @param \Propel\Runtime\ActiveQuery\ModelCriteria $existsQuery
+     * @param string|null $type Either ExistsCriterion::TYPE_EXISTS or ExistsCriterion::TYPE_NOT_EXISTS
+     * @param \Propel\Runtime\Map\RelationMap|null $relationMap where outer query is on the left side
+     */
+    public function __construct($outerQuery, $existsQuery, ?string $type = null, ?RelationMap $relationMap = null)
```

Comment:
Am I right that the `$type` parameter could be improved by having the default value be `self::TYPE_EXISTS` ? (looking at line 51 below)

Reply:
Since the input parameter can be any string, I have to check if it is one of the expected ones anyway. Think of `new ExistsCriterion($outer, $inner, 'EXERSTST')`. So line 51 has to stay there anyway, and this comes down to if you think null is a better or worse default value. I think better, as it has less overhead.